content/en/docs/reference/setup-tools/kubeadm/kubeadm-init.md

```diff
@@ -176,20 +176,23 @@ policy during rolling upgrades.
 : With this feature gate enabled, when joining a new control plane node, a new etcd member will be created
 as a learner and promoted to a voting member only after the etcd data are fully aligned.
 
-`PublicKeysECDSA`
-: Can be used to create a cluster that uses ECDSA certificates instead of the default RSA algorithm.
-Renewal of existing ECDSA certificates is also supported using `kubeadm certs renew`, but you cannot
-switch between the RSA and ECDSA algorithms on the fly or during upgrades. Kubernetes
-{{< skew currentVersion >}} has a bug where keys in generated kubeconfig files are set use RSA
-despite the feature gate being enabled. Kubernetes versions before v1.31 had a bug where keys in generated kubeconfig files
-were set use RSA, even when you had enabled the `PublicKeysECDSA` feature gate.
+`NodeLocalCRISocket`
+: With this feature gate enabled, kubeadm will consume the CRI socket for each node from the file
```

Comment:
What does “consume“ mean?
I guess it means that kubeadm parses that file and then works out what the path to the socket should be; have I got that right?

Ideally we document the format of this configuration file and update https://kubernetes.io/docs/reference/node/kubelet-files/ 

- if the kubelet does use that file, we should definitely update the reference
- if the kubelet ignores that file, but someone might find a file `/var/lib/kubelet/instance-config.yaml` on their node, we should give it a special mention within https://kubernetes.io/docs/reference/node/kubelet-files/  to explain the situation.

Reply:
> What does “consume“ mean?

consume == parse
i can change the verb.

> I guess it means that kubeadm parses that file and then works out what the path to the socket should be; have I got that right?

yes, since the file contents are a single line `containerRuntimeEndpoint: VALUE` i will add it in the FG text as an example.

> https://kubernetes.io/docs/reference/node/kubelet-files/

it's not a kubelet file and the kubelet does not know about it, so it should not be covered in that page.

Reply:
updated for more clarity.